core/src/character.rs

```diff
@@ -12,7 +12,10 @@ pub enum Character<'gc> {
     EditText(EditText<'gc>),
     Graphic(Graphic<'gc>),
     MovieClip(MovieClip<'gc>),
-    Bitmap(Bitmap<'gc>),
+    Bitmap {
+        bitmap: Bitmap<'gc>,
```

Comment:
I think that with #8316, we can store `ruffle_render::bitmap::Bitmap` directly in `ruffle_core::display_object::BitmapData`, and leave `Character::Bitmap` as-is. I see it nice to keep the `Character` enum with a single field per each member.

Reply:
We need to have the data available in `Character`, since it's possible to have a `Bitmap` with no associated `BitmapData`.

Reply:
But `Bitmap`s with no associated `BitmapData` cannot be mutated, right? If so, maybe we can "upgrade" the [`bitmap_data`](https://github.com/ruffle-rs/ruffle/blob/d0230a2beaa99dd067b94cf34d484f10545f3210/core/src/display_object/bitmap.rs#L61) field from `Option<GcCell<'gc, crate::bitmap::bitmap_data::BitmapData<'gc>>>` to an enum that can be either `GcCell<'gc, crate::bitmap::bitmap_data::BitmapData<'gc>>` or `ruffle_render::bitmap::Bitmap`?

Actually, `ruffle_render::bitmap::Bitmap` and `ruffle_core::bitmap::bitmap_data::BitmapData` seem very similar in their idea. Maybe we can somehow merge the two in the future.

Reply:
We only need the original pixels when creating a `DisplayObject::Bitmap` from an SWF character id. I think moving things into the `bitmap_data` field will just make things more complex, since the rest of the codebase really only cares about `BitmapData`.